layer4/routes.go

```diff
@@ -108,15 +108,14 @@ func (routes RouteList) Compile(logger *zap.Logger, matchingTimeout time.Duratio
 		if err != nil {
 			return err
 		}
-		isFirstPrefetch := true
-		for { // retry prefetching and matching routes until timeout
+
+		for i := 0; i < 10000; i++ { // retry prefetching and matching routes until timeout
 
 			// Do not call prefetch if this is the first loop iteration and there already is some data available,
 			// since this means we are at the start of a subroute handler and previous prefetch calls likely already fetched all bytes available from the client.
 			// Which means it would block the subroute handler. In the second iteration (if no subroute routes match) blocking is the correct behaviour.
-			if !isFirstPrefetch || cx.buf == nil || len(cx.buf[cx.offset:]) == 0 {
+			if i != 0 || cx.buf == nil || len(cx.buf[cx.offset:]) == 0 {
```

Comment:
Does this generalize well? It feels right but I just wanted to make sure that, in all cases, this should not be the first iteration.

Reply:
Yes if it's the first iteration and there is already data trying to match before another prefetch is best. Trying all matchers should be fast and we do not risk blocking.